flight/core/Dispatcher.php

```diff
@@ -105,42 +123,59 @@ public function has(string $name): bool
      */
     public function clear(?string $name = null): void
     {
-        if (null !== $name) {
+        if ($name !== null) {
             unset($this->events[$name]);
             unset($this->filters[$name]);
-        } else {
-            $this->events = [];
-            $this->filters = [];
+
+            return;
         }
+
+        $this->events = [];
+        $this->filters = [];
     }
 
     /**
      * Hooks a callback to an event.
      *
-     * @param string   $name     Event name
-     * @param string   $type     Filter type
-     * @param callable $callback Callback function
+     * @param string $name Event name
+     * @param 'before'|'after' $type Filter type
+     * @param Closure(array<int, mixed> &$params, string &$output): (void|false) $callback
+     *
+     * @return $this
      */
-    public function hook(string $name, string $type, callable $callback): void
+    public function hook(string $name, string $type, callable $callback): self
     {
+        if (!in_array($type, self::FILTER_TYPES, true)) {
+            $noticeMessage = "Invalid filter type '$type', use " . join('|', self::FILTER_TYPES);
+
+            trigger_error($noticeMessage, E_USER_NOTICE);
```

Comment:
This is another one of those things. I don't know if someone set up `'afterEmail'` as a hook in an old app, and they used it in a weird way outside of the framework, (which you can use cause a lot of these Dispatcher methods are public). I would say you'd have to revert this.

Reply:
After going through the unit test, I still don't know if this is something people could have overwritten in the past where they setup their custom hooks. I just don't know what I don't know :(

Reply:
> After going through the unit test, I still don't know if this is something people could have overwritten in the past where they setup their custom hooks. I just don't know what I don't know :(

Ah, I understand what you mean, by inheriting from the class you can increase the number of filters. Okay, I'll fix that.

Reply:
Although coming back to think, it doesn't make much sense to add more filters to an event, in the execution cycle There are only 2 moments of the method, before and after, and these can already be chained